pyked/chemked.py

```diff
@@ -718,7 +719,17 @@ def __init__(self, properties):
     def process_quantity(self, properties):
         """Process the uncertainty information from a given quantity and return it
         """
-        quant = Q_(properties[0])
+        try:
+            quant = Q_(properties[0])
+        except pint.OffsetUnitCalculusError:
+            values = properties[0].split()
+            quant = Q_(float(values[0]), ''.join(values[1:]))
+        except pint.UndefinedUnitError:
+            values = properties[0].split()
+            if values[0] == 'nan':
```

Comment:
In what situations do we end up in this branch? Why are we checking for nan specifically? All the values should be filled in from the YAML file, as far as I know. Any branches added here (if/else statements) need tests for both sides of the statement.

Reply:
I've seen some cases where the yaml file will have a value given as `nan`. In that case, pint gives the `UndefinedUnitError` and we end up in this branch, which will replace an ignition delay of `Q_('nan ms')` with `Q_(np.nan, 'ms')`. I added the if/else statements because there are other issues that raise this error. Note that I made a second commit (5c74873) that raises the `UndefinedUnitError` if it really is an undefined unit (as in a typo, `Q_('50.2 mms')`)

Reply:
@kyleniemeyer What are your thoughts on nan being specified in the yaml? I think those values should just not be specified

Reply:
I do not think we should allow nan in the YAML files—I can't really see a case where that would be necessary. If something wasn't detected (... first-stage ignition, maybe?) then the field should just not be specified.

@jsantner can you give a little more detail where you saw nan in use? Just because someone put it in doesn't mean we should support it... since currently that isn't part of the standard.

Reply:
In one of the RCM workshop yaml files, ignition delay was given as nan. I'm guessing somebody wrote a script to automatically find ignition delay and create the yaml file, and didn't catch an error in their ignition delay determination. Since you both don't think pyked should allow nan's, then the UndefinedUnitError doesn't need to be excepted here - it will be raised, and that should show a user that their 'nan' is a problem